src/main/java/hudson/plugins/git/GitChangeSet.java

```diff
@@ -412,14 +443,25 @@ public User findOrCreateUser(String csAuthor, String csAuthorEmail, boolean crea
         return user;
     }
 
+    private String getMail(User user) {
+        hudson.tasks.Mailer.UserProperty property = user.getProperty(hudson.tasks.Mailer.UserProperty.class);
+        if (property == null) {
+            return null;
+        }
+        if (!property.hasExplicitlyConfiguredAddress()) {
+            return null;
+        }
+        return property.getExplicitlyConfiguredAddress();
+    }
+
     private void setMail(User user, String csAuthorEmail) throws IOException {
         user.addProperty(new hudson.tasks.Mailer.UserProperty(csAuthorEmail));
     }
 
     private boolean hasMail(User user) {
-        hudson.tasks.Mailer.UserProperty property = user.getProperty(hudson.tasks.Mailer.UserProperty.class);
-        return property != null && property.hasExplicitlyConfiguredAddress();
-	}
+        String email = getMail(user);
+        return email != null && !email.isEmpty();
```

Comment:
Since you've added a new condition to return false if the property has an explicitly configured address that is the empty string, could you also add a test for that condition?

Was there a compelling reason that you added the new condition?  

Have you found cases where an explicitly configured address is the empty string?

Reply:
Reviewing the code for hasExplicitlyConfiguredAddress and getExplicitlyConfiguredAddress, an empty email will result in a null email adress.
So this is completely not needed, which means I can remove it, and probably added it for bad reasons.